ddl/partition.go

```diff
@@ -861,6 +861,21 @@ func getPartitionDef(tblInfo *model.TableInfo, partName string) (index int, def
 	return index, nil, table.ErrUnknownPartition.GenWithStackByArgs(partName, tblInfo.Name.O)
 }
 
+func buildPlacementDropRules(schemaID, tableID int64, partitionIDs []int64) []*placement.RuleOp {
+	rules := make([]*placement.RuleOp, 0, len(partitionIDs))
+	for _, partitionID := range partitionIDs {
+		rules = append(rules, &placement.RuleOp{
+			Action:           placement.RuleOpDel,
+			DeleteByIDPrefix: true,
+			Rule: &placement.Rule{
+				GroupID: placement.RuleDefaultGroupID,
+				ID:      fmt.Sprintf("%d_t%d_p%d", schemaID, tableID, partitionID),
```

Comment:
In ddl_api.go the rule ID is `rule.ID = fmt.Sprintf("%d_t%d_p%d_%s", job.SchemaID, tblInfo.ID, partitionID, rule.Role)`. Why here we needn't `rule.Role`?

Reply:
It drop all roles for this partition.

Reply:
Yes, it will delete all rules by this prefix.

Reply:
So why we use `rule.Role` in `onAlterTablePartition`? 
![WeChatWorkScreenshot_618e1877-624a-4fb2-a3e2-f4d488da0b43](https://user-images.githubusercontent.com/4242506/90712097-d607c900-e2d4-11ea-867b-da9119a3d344.png)

Reply:
Altering partition drops and adds a rule for a specific role, not all roles.

Reply:
Got it.